### PR TITLE
Fix for strictNullChecks & external package

### PR DESCRIPTION
**Pull request by katemihalikova**
- This PR makes the generated code compatible with `strictNullChecks: true`
- This PR allows the generated code to be in a separate npm package, fixes Microsoft/TypeScript#9944

### CODE REVIEW

File: demo-app/client/generated/store/career/positions/states/effects.ts [hunk 1]
```diff
@@ -9,16 +9,17 @@ import {HttpErrorResponse} from '@angular/common/http';
...
 import {catchError, map, switchMap} from 'rxjs/operators';
+
 import {CareerService} from '../../../../controllers/Career';
 import * as actions from './actions';
 
 @Injectable()
 export class PositionsEffects {
   @Effect()
-  Positions = this.storeActions.ofType<actions.Start>(actions.Actions.START).pipe(
+  Positions: Observable<actions.PositionsAction> = this.storeActions.ofType<actions.Start>(actions.Actions.START).pipe(
```
Comment (jnwltr) [inline demo-app/client/generated/store/career/positions/states/effects.ts]: @katemihalikova, please, could you explain why do we need this?
Originally, `Observable<actions.Success | actions.Error>` was inferred. What does this explicit typing bring? Thx.

File: demo-app/client/generated/store/career/positions/states/reducers.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ export const initialPositionsState: PositionsState = {
 };
 
 export const selectorName = 'Positions';
-export const getPositionsStateSelector = createFeatureSelector<PositionsState>(selectorName);
+export const getPositionsStateSelector: MemoizedSelector<object, PositionsState> = createFeatureSelector<PositionsState>(selectorName);
```
Comment (jnwltr) [inline demo-app/client/generated/store/career/positions/states/reducers.ts]: Probably same reason as above, right?

**Comment by jnwltr** [comment]
@katemihalikova, could you, please, explain concretely on our case the second point (generated code in separate npm package)? Thx

**Comment by katemihalikova** [comment]
Sure. We are having a separate package with just the generated code that is build using ngc separately from the app. Because of Microsoft/TypeScript#9944, anything related to exported types and properties that originates in a dependency (like `Observable`) must be imported to make the build working.

This is done with e.g. `import {Observable} from 'rxjs/Observable';`.

But because it is still desirable to support `"noUnusedLocals": true` compile option, the imported type must be used somewhere so I decided to make the type explicit. That means that the type is correctly inferred, but is added just to fix the `"noUnusedLocals": true` problem.

**Comment by jnwltr** [comment]
I suggest to abandon this one for now. Do you agree?

**Pull request closed without merging**